transmogrifier/sources/xml/ead.py

```diff
@@ -425,6 +374,147 @@ def get_content_type(cls, source_record: Tag) -> list[str] | None:
 
         return content_types or None
 
+    @classmethod
+    def get_contents(cls, source_record: Tag) -> list[str] | None:
+        contents = []
+        collection_description = cls._get_collection_description(source_record)
+        for arrangement_element in collection_description.find_all(
+            "arrangement", recursive=False
+        ):
+            contents.extend(
+                cls.create_list_from_mixed_value(arrangement_element, ["head"])
+            )
+        return contents or None
+
+    @classmethod
+    def get_contributors(cls, source_record: Tag) -> list[timdex.Contributor] | None:
+        contributors = []
+        collection_description_did = cls._get_collection_description_did(source_record)
+        for origination_element in collection_description_did.find_all("origination"):
+            contributors.extend(
+                [
+                    timdex.Contributor(
+                        value=contributor_name,
+                        kind=origination_element.get("label") or None,
+                        identifier=cls._get_contributor_identifier_url(
+                            contributor_element
+                        ),
+                    )
+                    for contributor_element in origination_element.find_all(
+                        name=True, recursive=False
+                    )
+                    if (
+                        contributor_name := cls.create_string_from_mixed_value(
+                            contributor_element, " "
+                        )
+                    )
+                ]
+            )
+
+        return contributors or None
+
+    @classmethod
+    def _get_contributor_identifier_url(cls, name_element: Tag) -> list | None:
+        """Generate a full name identifier URL with the specified scheme.
+
+        Args:
+            name_element: A BeautifulSoup Tag representing an EAD
+                name XML field.
+        """
+        if identifier := name_element.get("authfilenumber"):
+            source = name_element.get("source")
+            if source in ["lcnaf", "naf"]:
+                base_url = "https://lccn.loc.gov/"
+            elif source == "snac":
+                base_url = "https://snaccooperative.org/view/"
+            elif source == "viaf":
+                base_url = "http://viaf.org/viaf/"
+            else:
+                base_url = ""
+            return [base_url + identifier]
+        return None
+
+    @classmethod
+    def get_dates(cls, source_record: Tag) -> list[timdex.Date] | None:
+        dates = []
+        collection_description_did = cls._get_collection_description_did(source_record)
+        source_record_id = cls.get_source_record_id(source_record)
+        dates.extend(
+            cls._parse_date_elements(collection_description_did, source_record_id)
+        )
```

Comment:
I know we're not addressing this now, but just want to continue to note that pulling the `source_record_id`, just so it can be logged for bad date parsing, feels awkward.

Apologies if an issue or note exists elsewhere, but just in case, created this one: https://github.com/MITLibraries/transmogrifier/issues/188.

Reply:
Agreed and thanks for creating the issue!